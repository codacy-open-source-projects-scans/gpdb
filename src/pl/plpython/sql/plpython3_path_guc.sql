--
-- Tests for functions that return set PYTHONPATH guc only for gpdb7 plpython3
--
show plpython3.python_path;

set plpython3.python_path='/foo';
show plpython3.python_path;

-- Set up test path functions first.
CREATE FUNCTION test_path_added() 
RETURNS text AS $$
  import sys
  return str(sys.path[0])
$$ language plpython3u;

SELECT test_path_added();

-- when the plpython init can not set again --
set plpython3.python_path='/bar';

-- run again the path will not change still /foo --
SELECT test_path_added();
