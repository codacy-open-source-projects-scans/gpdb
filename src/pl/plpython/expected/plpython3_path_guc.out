--
-- Tests for functions that return set PYTHONPATH guc only for gpdb7 plpython3
--
show plpython3.python_path;
 plpython3.python_path 
-----------------------
 
(1 row)

set plpython3.python_path='/foo';
show plpython3.python_path;
 plpython3.python_path 
-----------------------
 /foo
(1 row)

-- Set up test path functions first.
CREATE FUNCTION test_path_added() 
RETURNS text AS $$
  import sys
  return str(sys.path[0])
$$ language plpython3u;
SELECT test_path_added();
 test_path_added 
-----------------
 /foo
(1 row)

-- when the plpython init can not set again --
set plpython3.python_path='/bar';
ERROR:  SET PYTHONPATH failed, the GUC value can only be changed before initializing the python interpreter.
-- run again the path will not change still /foo --
SELECT test_path_added();
 test_path_added 
-----------------
 /foo
(1 row)

